File: Top 50 Coding Interview Questions/38 Write a Program to Find out the Sum of Digits of a Number/FindouttheSumofDigitsofaNumber.cpp
#include <iostream> // Include library for input/output operations
using namespace std;

int main() {
    int num, sum = 0; // Declare variables: num to store input, sum to store result

    // Prompt user for input
    cout << "Enter any number: ";
    cin >> num;

    // Loop to calculate the sum of digits
    while (num != 0) {
        sum += num % 10;  // Add the last digit of num to sum
        num /= 10;        // Remove the last digit from num
    }

    // Output the result
    cout << "\nSum of digits: " << sum << endl;

    return 0;
}

/*
Time Complexity: O(N) 
- Where N is the number of digits in the input number.
- The loop iterates once for each digit.

Space Complexity: O(1)
- Only a constant amount of extra memory is used (variables num and sum).
*/

File: Top 50 Coding Interview Questions/40 Write a Program to Find out the Sum of Digits of a Number/FindouttheSumofDigitsofaNumber.py
# Prompt the user to input a number as a string
num = input("Enter Number: ")

# Initialize the variable 'sum' to store the sum of the digits
sum = 0

# Loop through each character in the input string
for i in num:
    # Convert the character to an integer and add it to the sum
    sum = sum + int(i)

# Print the calculated sum of the digits
print(sum)
